In [181]:
import pandas as pd

In [182]:
#Load File
file_to_load = "../Resources/purchase_data.csv"

In [183]:
#Read File
purchase_data = pd.read_csv(file_to_load)
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [184]:
#Unique Player Count
unique_sn_count = purchase_data['SN'].nunique()
unique_sn_count

576

In [185]:
#Unique Game Count
unique_item_count = purchase_data['Item Name'].nunique()
#unique_item_count

#Average Purchase Price
average_price = purchase_data['Price'].mean()
#average_price

#Total Purchases
total_purchases = purchase_data['Purchase ID'].count()
#total_purchases

#Total Revenue
total_revenue = purchase_data['Price'].sum()
#total_revenue

In [186]:
#Purchasing Analysis (Total)

pd.DataFrame([{
    
    'Number of Unique Items': unique_item_count,
    'Average Purchase Price': '${:,.2f}'.format(average_price),
    'Total Number of Purchases': total_purchases,
    'Total Revenue': '${:,.2f}'.format(total_revenue)
    
}])

Number of Unique Items Average Purchase Price  Total Number of Purchases  \
0                     179                  $3.05                        780   

  Total Revenue  
0     $2,379.77

In [187]:
#Gender Demographics
player_demographics = purchase_data.loc[:, ["Gender", "SN", "Age"]]
player_demographics = player_demographics.drop_duplicates()
gender_demographics_totals = player_demographics["Gender"].value_counts()

gender_percentage = gender_demographics_totals / unique_sn_count

In [188]:
#Gender Demographics DataFrame
gender_summary = pd.DataFrame({'Total Count': gender_demographics_totals,
                               '% of Players': gender_percentage
                             })
gender_summary

Total Count  % of Players
Male                           484      0.840278
Female                          81      0.140625
Other / Non-Disclosed           11      0.019097

In [189]:
#Unremove removal of Duplicates
player_demographics = purchase_data.loc[:, ["Gender", "SN", "Age"]]
gender_demographics_totals_full = player_demographics["Gender"].value_counts()
#gender_demographics_totals_full

#Purchase Value Total
purch_value_total = purchase_data.groupby(["Gender"]).sum()["Price"]
#purch_value_total

#Avrage Purchase Price
avg_purch_price = purchase_data.groupby(["Gender"]).mean()["Price"]
avg_purch_price



Gender
Female                   3.203009
Male                     3.017853
Other / Non-Disclosed    3.346000
Name: Price, dtype: float64

In [191]:
#Gender Demographics DataFrame
value_summary = pd.DataFrame({'Total Count': gender_demographics_totals_full,
                              'Average Purchase Price': avg_purch_price,
                              'Total Purchase Value': purch_value_total,
                            })
value_summary

Total Count  Average Purchase Price  \
Female                         113                3.203009   
Male                           652                3.017853   
Other / Non-Disclosed           15                3.346000   

                       Total Purchase Value  
Female                               361.94  
Male                                1967.64  
Other / Non-Disclosed                 50.19

In [192]:
#Create Bins
bins = [0, 9, 14, 19, 24, 29, 34, 39, 99]

age_labels = ['<10', '10-14', '15-19', '20-24', '25-29', '30-34', '35-39', '40-99']

In [193]:
#Slice the Data and put it into Bins
view_group = pd.cut(purchase_data['Age'], bins, labels=age_labels)
view_group

0      20-24
1      40-99
2      20-24
3      20-24
4      20-24
       ...  
775    20-24
776    20-24
777    20-24
778      <10
779    20-24
Name: Age, Length: 780, dtype: category
Categories (8, object): [<10 < 10-14 < 15-19 < 20-24 < 25-29 < 30-34 < 35-39 < 40-99]

In [194]:
# Place the data series into a new column inside of the DataFrame
purchase_data['View Group'] = view_group
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price View Group  
0  Extraction, Quickblade Of Trembling Hands   3.53      20-24  
1                          Frenzied Scimitar   1.56      40-99  
2                               Final Critic   4.88      20-24  
3                                Blindscythe   3.27      20-24  
4                                       Fury   1.44      20-24

In [195]:
#Create groupby object based on Age
age_group = purchase_data.groupby(['View Group'])

#Count Age Groups
age_group.count()['SN']

View Group
<10       23
10-14     28
15-19    136
20-24    365
25-29    101
30-34     73
35-39     41
40-99     13
Name: SN, dtype: int64

In [196]:
#Age Summary Total
age_summary_totals = purchase_data.groupby(["View Group"]).count()["Age"]

percent_of_players = age_summary_totals / unique_sn_count

In [197]:
#Create Data Frame
age_summary = pd.DataFrame({'Total Count': age_summary_totals,
                            'Percentage of Players': percent_of_players
                            })
age_summary

Total Count  Percentage of Players
View Group                                    
<10                  23               0.039931
10-14                28               0.048611
15-19               136               0.236111
20-24               365               0.633681
25-29               101               0.175347
30-34                73               0.126736
35-39                41               0.071181
40-99                13               0.022569

In [198]:
#Calculations for Data Frame
avg_age_purch_price = purchase_data.groupby(["View Group"]).mean()["Price"]

age_purch_value = purchase_data.groupby(["View Group"]).sum()["Price"]

avg_tot_per_person = age_purch_value / age_summary_totals

In [199]:
#Create Data Fram
age_analysis = pd.DataFrame({'Purchase Count': age_summary_totals,
                             'Average Purchase Price': avg_age_purch_price,
                             'Total Purchase Value': age_purch_value,
                             'Avg Total Purch Per Person': avg_tot_per_person
                            })
age_analysis

Purchase Count  Average Purchase Price  Total Purchase Value  \
View Group                                                                 
<10                     23                3.353478                 77.13   
10-14                   28                2.956429                 82.78   
15-19                  136                3.035956                412.89   
20-24                  365                3.052219               1114.06   
25-29                  101                2.900990                293.00   
30-34                   73                2.931507                214.00   
35-39                   41                3.601707                147.67   
40-99                   13                2.941538                 38.24   

            Avg Total Purch Per Person  
View Group                              
<10                           3.353478  
10-14                         2.956429  
15-19                         3.035956  
20-24                         3.052219  
25-29                         2.900990  
30-34                         2.931507  
35-39                         3.601707  
40-99                         2.941538

In [200]:
#Calculations for SN

sn_df = purchase_data.loc[:, ["SN", "Item Name", "Price"]]
sn_df_count = sn_df["SN"].value_counts()
#sn_df_count

sn_avg_purchase_price = purchase_data.groupby(["SN"]).mean()["Price"]
#sn_avg_purchase_price

sn_total_purchases = purchase_data.groupby(["SN"]).sum()["Price"]
sn_total_purchases

SN
Adairialis76     2.28
Adastirin33      4.48
Aeda94           4.91
Aela59           4.32
Aelaria33        1.79
                 ... 
Yathecal82       6.22
Yathedeu43       6.02
Yoishirrala98    4.58
Zhisrisu83       7.89
Zontibe81        8.03
Name: Price, Length: 576, dtype: float64

In [201]:
#Create Data Frame
sn_analysis = pd.DataFrame({
    
                              'Purchase Count': sn_df_count,
                              'Average Purchase Price': sn_avg_purchase_price,
                              'Total Purchase Value': sn_total_purchases
                            })
sn_analysis

Purchase Count  Average Purchase Price  Total Purchase Value
Adairialis76                1                2.280000                  2.28
Adastirin33                 1                4.480000                  4.48
Aeda94                      1                4.910000                  4.91
Aela59                      1                4.320000                  4.32
Aelaria33                   1                1.790000                  1.79
...                       ...                     ...                   ...
Yathecal82                  3                2.073333                  6.22
Yathedeu43                  2                3.010000                  6.02
Yoishirrala98               1                4.580000                  4.58
Zhisrisu83                  2                3.945000                  7.89
Zontibe81                   3                2.676667                  8.03

[576 rows x 3 columns]

In [206]:
sn_sort = sn_analysis.sort_values('Total Purchase Value', ascending=False)
sn_sort.head()

Purchase Count  Average Purchase Price  Total Purchase Value
Lisosia93                 5                3.792000                 18.96
Idastidru52               4                3.862500                 15.45
Chamjask73                3                4.610000                 13.83
Iral74                    4                3.405000                 13.62
Iskadarya95               3                4.366667                 13.10

In [202]:
#Calculations for Popular / Profitable Data Frame

item_df = purchase_data.loc[:, ["Item ID", "Item Name", "Price"]]
item_df_count = item_df["Item ID"].value_counts()
#item_df_count

game_purchase_value = purchase_data.groupby(["Item ID"]).sum()["Price"]
#game_purchase_value

game_item_price = purchase_data.groupby(["Item ID"]).mean()["Price"]
game_item_price


Item ID
0      1.2800
1      2.9425
2      2.4800
3      2.4900
4      1.7000
        ...  
178    4.2300
179    4.4800
181    1.6600
182    4.0300
183    1.0900
Name: Price, Length: 179, dtype: float64

In [203]:
#Create Data Frame
item_analysis = pd.DataFrame({
    
                              'Item Name': item_df['Item Name'],
                              'Purchase Count': item_df_count,
                              'Item Price': game_item_price,
                              'Total Purchase Value': game_purchase_value
                            })
item_analysis

Item Name  Purchase Count  Item Price  \
0    Extraction, Quickblade Of Trembling Hands             4.0      1.2800   
1                            Frenzied Scimitar             4.0      2.9425   
2                                 Final Critic             6.0      2.4800   
3                                  Blindscythe             6.0      2.4900   
4                                         Fury             5.0      1.7000   
..                                         ...             ...         ...   
775                                       Wolf             NaN         NaN   
776                           Exiled Doomblade             NaN         NaN   
777      Celeste, Incarnation of the Corrupted             NaN         NaN   
778                               Final Critic             NaN         NaN   
779                                       Dawn             NaN         NaN   

     Total Purchase Value  
0                    5.12  
1                   11.77  
2                   14.88  
3                   14.94  
4                    8.50  
..                    ...  
775                   NaN  
776                   NaN  
777                   NaN  
778                   NaN  
779                   NaN  

[780 rows x 4 columns]

In [204]:
item_sort = item_analysis.sort_values('Purchase Count', ascending=False)
item_sort.head()

Item Name  Purchase Count  Item Price  \
92   Betrayal, Whisper of Grieving Widows            13.0    4.614615   
178       Despair, Favor of Due Diligence            12.0    4.230000   
145               Hopeless Ebon Dualblade             9.0    4.580000   
132                  Fiery Glass Crusader             9.0    3.221111   
108                        Malificent Bag             9.0    3.530000   

     Total Purchase Value  
92                  59.99  
178                 50.76  
145                 41.22  
132                 28.99  
108                 31.77

In [205]:
item_sort = item_analysis.sort_values('Total Purchase Value', ascending=False)
item_sort.head()

Item Name  Purchase Count  Item Price  \
92   Betrayal, Whisper of Grieving Widows            13.0    4.614615   
178       Despair, Favor of Due Diligence            12.0    4.230000   
82                             Azurewrath             9.0    4.900000   
145               Hopeless Ebon Dualblade             9.0    4.580000   
103          Thorn, Satchel of Dark Souls             8.0    4.350000   

     Total Purchase Value  
92                  59.99  
178                 50.76  
82                  44.10  
145                 41.22  
103                 34.80

Three Observable Trends

1) Most gamers are in the 20-24 age group
2) Item 92 and 178 are the biggest sellers
3) Great majority of gamers are Male